src/components/Feature.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import FlagContext from './FlagContext';
+import Config from '../config';
+
+class Feature extends Component {
+  render() {
+    const { children, flagId, variation, flags } = this.props;
+
+    const renderNode =
+      variation !== undefined
+        ? flags[flagId] === variation && children
+        : children;
+
+    return (
+      <FlagContext.Provider value={ flags[flagId] }>
+        {renderNode}
+      </FlagContext.Provider>
+    );
+  }
+}
+
+Feature.propTypes = {
+  children: PropTypes.node.isRequired,
+  flagId: PropTypes.string.isRequired,
+  variation: PropTypes.any,
```

Comment:
any? not required? What is the use case of not having a variation?

Reply:
it's any because it can be a string, null, or boolean.

The case of not having a variation prop on the `Feature` component is when you are using the `Variant` component as a child of `Feature`, at which point `Feature` simply provides the context of the feature flag for the `Variant` components.

The only reason `Feature` has a `variation` prop is so you don't have to needlessly use a `Variant` component as a child of `Feature` if you only care about one variant. (Look at the `/example/test-app/appStuff/components` to see what I mean)